tests/prague/eip7685_general_purpose_el_requests/conftest.py

```diff
@@ -50,6 +58,26 @@ def exception() -> BlockException | None:
     return None
 
 
+@pytest.fixture
+def engine_api_error_code(
+    block_body_override_requests: List[Bytes | SupportsBytes] | None,
+) -> EngineAPIError | None:
+    """Engine API error code if any."""
+    if block_body_override_requests is None:
+        return None
+    block_body_override_requests_bytes = [bytes(r) for r in block_body_override_requests]
+    if any(len(r) <= 1 for r in block_body_override_requests_bytes):
+        return EngineAPIError.InvalidParams
+
```

Comment:
And don't we need this?
```suggestion

        if not all(0 < r[0] < fork.max_request_type() for r in block_body_override_requests_bytes):
        return EngineAPIError.InvalidParams

```

Reply:
This change didn't change any fixtures from `tests/prague/eip7685_general_purpose_el_requests/test_request_types.py::test_invalid_request_type`.

But added the this engine api error field:
```json
                "errorCode": "-32602"
```
 to the following fixtures:
```
tests/prague/eip7685_general_purpose_el_requests/test_deposits_withdrawals_consolidations.py::test_invalid_deposit_withdrawal_consolidation_requests[fork_Prague-single_consolidation_missing_type_byte-blockchain_test_engine]
tests/prague/eip7685_general_purpose_el_requests/test_deposits_withdrawals_consolidations.py::test_invalid_deposit_withdrawal_consolidation_requests[fork_Prague-single_deposit_missing_type_byte-blockchain_test_engine]
tests/prague/eip7685_general_purpose_el_requests/test_deposits_withdrawals_consolidations.py::test_invalid_deposit_withdrawal_consolidation_requests[fork_Prague-single_withdrawal_missing_type_byte-blockchain_test_engine]
tests/prague/eip7685_general_purpose_el_requests/test_deposits_withdrawals_consolidations.py::test_invalid_deposit_withdrawal_consolidation_requests_engine[fork_Prague-single_consolidation_missing_type_byte-blockchain_test_engine-correct_requests_hash_in_header_True]
tests/prague/eip7685_general_purpose_el_requests/test_deposits_withdrawals_consolidations.py::test_invalid_deposit_withdrawal_consolidation_requests_engine[fork_Prague-single_deposit_missing_type_byte-blockchain_test_engine-correct_requests_hash_in_header_True]
tests/prague/eip7685_general_purpose_el_requests/test_deposits_withdrawals_consolidations.py::test_invalid_deposit_withdrawal_consolidation_requests_engine[fork_Prague-single_withdrawal_missing_type_byte-blockchain_test_engine-correct_requests_hash_in_header_True]
```

Reply:
Thanks for the clarification: An invalid request type should not cause an EngineAPIError, rather an invalid block, which these test already do.

Reply:
This one is not necessary because the execution-apis spec does not include invalid request types as cause for an Engine API error, and my reasoning behind it was that this check requires knowledge about the fork in which the block is being processed, so it could be a misconfiguration of the CL fork schedule rather than a Engine API error.